### PR TITLE
bpo-41879: Doc: Fix description of async for statement

### PR DESCRIPTION
**Pull request by nickgaya**
Fix the wording in the documentation of `async for` to correctly describe asynchronous iterables.  This fix is relevant for version 3.7 onward.

https://bugs.python.org/issue41879

Automerge-Triggered-By: GH:aeros

### CODE REVIEW

**Comment by the-knights-who-say-ni** [comment]
Hello, and thanks for your contribution!

I'm a bot set up to make sure that the project can legally accept this contribution by verifying everyone involved has signed the [PSF contributor agreement](https://www.python.org/psf/contrib/contrib-form/) (CLA).



## CLA Missing

Our records indicate the following people have not signed the CLA:

@nickgaya

For legal reasons we need all the people listed to sign the CLA before we can look at your contribution. Please follow [the steps outlined in the CPython devguide](https://devguide.python.org/pullrequest/#licensing) to rectify this issue.

If you have recently signed the CLA, please wait at least one business day
before our records are updated.


You can [check yourself](https://check-python-cla.herokuapp.com/) to see if the CLA has been received.

Thanks again for the contribution, we look forward to reviewing it!

File: Doc/reference/compound_stmts.rst [hunk 1]
```diff
@@ -796,12 +796,12 @@ The :keyword:`!async for` statement
 .. productionlist:: python-grammar
    async_for_stmt: "async" `for_stmt`
 
-An :term:`asynchronous iterable` is able to call asynchronous code in its
-*iter* implementation, and :term:`asynchronous iterator` can call asynchronous
-code in its *next* method.
+An :term:`asynchronous iterable` provides an *iter* method that returns an
```
Comment (asvetlov) [inline Doc/reference/compound_stmts.rst]: Technically `*iter*` is `__aiter__` and `*next*` is `__anext__`.

I have no idea how to write the text better.  Maybe it should not name iter/next at all but define async-iterable/async-iterator in plain English words?

Comment (nickgaya) [reply Doc/reference/compound_stmts.rst]: I agree it's a bit unclear, but consistent with how async context managers are described in the `async with` section.

Comment (asvetlov) [reply Doc/reference/compound_stmts.rst]: I believe `async for` description should be compatible with `for`, `async with` should follow `with` structure.

@1st1 @aeros please share your opinions.

Comment (JulienPalard) [reply Doc/reference/compound_stmts.rst]: Beware english not my primary language, but here's my proposed wording:

    An :term:`asynchronous iterable` provides an ``__aiter__`` method that directly returns an
    :term:`asynchronous iterator`, which can call asynchronous code in its ``async __anext__``
    method.

Comment (asvetlov) [reply Doc/reference/compound_stmts.rst]: Looks good.

@1st1 @aeros maybe you can review the proposal?

Comment (1st1) [reply Doc/reference/compound_stmts.rst]: I like Julien's phrasing.

Comment (aeros) [reply Doc/reference/compound_stmts.rst]: +1 from me to go with Julian's proposal.

Comment (nickgaya) [reply Doc/reference/compound_stmts.rst]: I'm not too keen on this change as I think the reason for using italicized words originally was to refer to the methods without getting into the specifics, which are described in more detail in the data model documentation for asynchronous iterators.  Compare to the `async with` section, which introduces the concept of an asynchronous context manager as "a context manager that is able to suspend execution in its *enter* and *exit* methods" rather than using the exact names.

However Julien's proposal seems to have majority support so I updated the PR accordingly.

Comment (aeros) [reply Doc/reference/compound_stmts.rst]: The reference section of the docs is typically intended to be more technically accurate, rather than prioritizing a simplified introduction. I wouldn't be opposed to applying the same to `async with` (to instead refer to `__aexit__` and `__aenter__`), but don't think it's worth changing by itself.

**Review by aeros — APPROVED** [review]

**Comment by aeros** [comment]
By the way, thanks for the PR and welcome @nickgaya!

**Comment by aeros** [comment]
Adding backport label only for 3.9, as it's the latest version present on docs.python.org. I typically don't backport lower than that for documentation PRs, since there's minimal to no benefit for most users.
